internal/worker/apiremotecaller/remote.go

```diff
@@ -229,16 +236,21 @@ func (w *remoteServer) loop() error {
 				rctx      = request.ctx
 			)
 
-			w.logger.Debugf(ctx, "addresses for %q have changed: %v", w.controllerID, addresses)
+			w.logger.Debugf(ctx, "addresses for %q have changed: %v",
+				w.controllerID, addresses)
 
 			// If the addresses already exist, we don't need to do anything.
 			if connected && w.addressesAlreadyExist(addresses) {
-				w.logger.Debugf(ctx, "addresses for %q have not changed", w.controllerID)
+				w.logger.Debugf(ctx, "addresses for %q have not changed",
+					w.controllerID)
 				continue
+			} else {
+				w.logger.Debugf(ctx, "same addresses for %q", w.controllerID)
```

Comment:
This log is wrong. It would be good to just remove it.

I think this is a better improvement to the logs:
```go
var (
	addresses = request.addresses
	rctx      = request.ctx
)

if connected && w.addressesAlreadyExist(addresses) {
	w.logger.Debugf(ctx, "already connected with existing addresses for %q",
		w.controllerID)
	continue
}
w.logger.Debugf(ctx, "addresses for %q have changed: %v",
	w.controllerID, addresses)
```